tests/native/test_integration_native.py

```diff
@@ -260,6 +260,24 @@ def test_basic_arm(self):
         with open(os.path.join(workspace, "test_00000001.stdout")) as f:
             self.assertIn("Message", f.read())
 
+    def test_fclose_linux_amd64(self):
+        """
+        Tests that the fclose example for amd64 linux doesn't crash; see #1602 and #1604.
+        """
+        filename = os.path.abspath(os.path.join(DIRPATH, "binaries", "fclose_linux_amd64"))
+        workspace = os.path.join(self.test_dir, "workspace")
+        cmd = [
+            PYTHON_BIN,
+            "-m",
+            "manticore",
+            "--no-color",
+            "--workspace",
+            workspace,
+            filename,
+            "+++++++",
+        ]
+        subprocess.check_call(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
```

Comment:
I know the other tests are written using `os.path`, but I think we should try to modernize where/when possible and use `pathlib.Path`.

The only clunky thing might be that we have to convert the `Path` object to a `str` in the `cmd` list.

Reply:
For a (rather clunky) example, you can look at the WASM example tests: https://github.com/trailofbits/manticore/blob/934a4f0e332dfdbbe225a3f978a508429d5472a9/tests/wasm/test_examples.py#L35-L37

Reply:
The rest of the file uses `os.path` exclusively.  Switching just my newly-added test to use a different path manipulation API is going to make it more difficult for future readers -- they'll have to reason about 2 APIs instead of 1, and also wonder why the existing convention was broken in just one place in a file.

If we're gonna switch to a different API for path manipulation, I suggest we do it file-at-a-time as the smallest granularity.

Reply:
Good points 👍